osu.Game/Screens/Edit/Compose/Components/BlueprintContainer.cs

```diff
@@ -432,13 +447,13 @@ private bool endClickSelection(MouseButtonEvent e)
 
                 // The depth of blueprints is constantly changing (see above where selected blueprints are brought to the front).
                 // For this logic, we want a stable sort order so we can correctly cycle, thus using the blueprintMap instead.
-                IEnumerable<SelectionBlueprint<T>> cyclingSelectionBlueprints = blueprintMap.Values;
+                IEnumerable<SelectionBlueprint<T>> cyclingSelectionBlueprints = ApplySelectionOrder(blueprintMap.Values);
 
                 // If there's already a selection, let's start from the blueprint after the selection.
                 cyclingSelectionBlueprints = cyclingSelectionBlueprints.SkipWhile(b => !b.IsSelected).Skip(1);
 
                 // Add the blueprints from before the selection to the end of the enumerable to allow for cyclic selection.
-                cyclingSelectionBlueprints = cyclingSelectionBlueprints.Concat(blueprintMap.Values.TakeWhile(b => !b.IsSelected));
+                cyclingSelectionBlueprints = cyclingSelectionBlueprints.Concat(ApplySelectionOrder(blueprintMap.Values).TakeWhile(b => !b.IsSelected));
```

Comment:
It's probably a micro-optimisation, and only probably becomes a relatively relevant issue in https://github.com/ppy/osu/pull/24291, but I'm not sure how to feel about `ApplySelectionOrder()` getting called twice with the same argument and then having half the sorted items discarded every time. Maybe better to materialise the collection and use that instead?

Reply:
I dunno. One would hope this is never going to be iterating over more than 5 or 6 notes (they need to be visible and at the same screen location) so I didn’t over-think it.

Reply:
I'm _relatively_ sure that `blueprintMap` will have an entry for every single hitobject in the map, not just the alive ones...? Which is why I brought it up to begin with.

https://github.com/ppy/osu/blob/285a232cc20293f281da6a85e1f69a3d1c9da757/osu.Game/Screens/Edit/Compose/Components/EditorBlueprintContainer.cs#L50

Reply:
Right. I will change this to use `AliveObjects` or something else.

Reply:
Actually due to the structure that is really not feasible. If you feel strongly about this, please profile a case you think will be slow. I don't think it's worth worrying about (and I'm not confident materialising an array is going to be much faster, especially since the second usage may never be hit in the LINQ execution as it's only when a cycle happens).

Reply:
Profiled an extreme case which is basically: take a marathon map, make 2 notes overlap, click on intersection to toggle selection (as stated above requires https://github.com/ppy/osu/pull/24291):

https://github.com/ppy/osu/assets/20418176/6338e0a4-63ff-4501-a54f-199326b8f372

Results:

| current code | with `.ToArray()` |
| :-: | :-: |
| ![1689872466](https://github.com/ppy/osu/assets/20418176/d5be95d2-5ef4-46c3-a129-59cf1765b462) | ![1689872475](https://github.com/ppy/osu/assets/20418176/8a59c553-7e87-4733-90b9-de7da1120bcd) |

It's _faster_, but not sure if fast enough to make worth. I guess we can probably go with the slow version unless this becomes an actual issue? dunno.

Reply:
Quite marginal. Up to you. I think this could probably be fixed in a better way, only edge case is skin editor, where we’d really want a Drawable.ID for stability..

Reply:
I'm just gonna go along with this as is to push the series along. If in the worst case the array dump only yields like 0.15ms per call then it's probably not worth pulling hair over..